src/transforms/sample/transform.rs

```diff
@@ -12,15 +12,44 @@ use crate::{
 use vector_lib::lookup::lookup_v2::OptionalValuePath;
 use vector_lib::lookup::OwnedTargetPath;
 
+/// Exists only for backwards compatability purposes so that the value of sample_rate_key is
+/// consistent after the internal implementation of the Sample class was modified to work in terms
+/// of percentages
+#[derive(Clone, Copy, Debug)]
+pub enum SampleRate {
+    OneOverN(u64),
+    Percentage(f32),
+}
+
+impl SampleRate {
+    pub fn pct(&self) -> f32 {
+        match self {
+            // Potential precision loss if user had a very large number in old rate implementation
+            Self::OneOverN(integer_rate) => 1.0 / (*integer_rate as f32),
```

Comment:
Is this calculating the percentage or a ratio? If the former, it should be `100.0` here. I would, however, argue for using ratios to avoid any issues with scaling by 100.